From dd6c037aa54f5a58cc7f2bb4a9f66f1b308f4cf4 Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Sat, 24 Nov 2018 19:12:44 +0800
Subject: [PATCH 8/9] rtc:pcf2127:fix a kmemleak issue.

Fix following kmemleak issue:
unreferenced object 0xffff800964b4ba00 (size 64):
  comm "kworker/1:1", pid 32303, jiffies 4298592256 (age 1227.268s)
  hex dump (first 32 bytes):
    03 45 52 08 23 05 11 18 08 ba b4 64 09 80 ff ff .ER.#......d....
    08 ba b4 64 09 80 ff ff 00 00 00 00 00 00 00 00 ...d............
  backtrace:
    [<ffff000008230688>] create_object+0xf8/0x278
    [<ffff000008af4adc>] kmemleak_alloc+0x74/0xa0
    [<ffff00000821f9fc>] __kmalloc+0x1ac/0x348
    [<ffff00000887d5cc>] pcf2127_i2c_gather_write+0x54/0xf8
    [<ffff00000866857c>] _regmap_raw_write+0x464/0x850
    [<ffff000008668f9c>] regmap_bulk_write+0x1a4/0x348
    [<ffff00000887d71c>] pcf2127_rtc_set_time+0xac/0xe8
    [<ffff00000887aec8>] rtc_set_time+0x80/0x138
    [<ffff000008879f14>] rtc_set_ntp_time+0x74/0xc8
    [<ffff0000081196c8>] sync_cmos_clock+0x90/0x188
    [<ffff0000080c5240>] process_one_work+0x1f8/0x4a8
    [<ffff0000080c5540>] worker_thread+0x50/0x478
    [<ffff0000080cbca8>] kthread+0x138/0x140
    [<ffff0000080854a0>] ret_from_fork+0x10/0x18
    [<ffffffffffffffff>] 0xffffffffffffffff

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/rtc/rtc-pcf2127.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/rtc/rtc-pcf2127.c b/drivers/rtc/rtc-pcf2127.c
index f33447c..78fa360 100644
--- a/drivers/rtc/rtc-pcf2127.c
+++ b/drivers/rtc/rtc-pcf2127.c
@@ -248,6 +248,7 @@ static int pcf2127_i2c_gather_write(void *context,
 	memcpy(buf + 1, val, val_size);
 
 	ret = i2c_master_send(client, buf, val_size + 1);
+	kfree(buf);
 	if (ret != val_size + 1)
 		return ret < 0 ? ret : -EIO;
 
-- 
1.7.9.5

